contracts/FlowIDTableStaking.cdc

```diff
@@ -768,10 +768,16 @@ pub contract FlowIDTableStaking {
         /// Sets a list of approved node IDs for the next epoch
         /// Nodes not on this list will be unstaked at the end of the staking auction
         /// and not considered to be a proposed/staked node
-        pub fun setApprovedList(_ nodeIDs: [String]) {
-            let list = FlowIDTableStaking.account.load<[String]>(from: /storage/idTableApproveList)
+        pub fun setApprovedList(_ nodeIDs: {String: Bool}) {
+            for id in nodeIDs.keys {
+                if FlowIDTableStaking.nodes[id] == nil {
+                    panic("Approved node ".concat(id).concat(" does not already exist in the identity table"))
```

Comment:
This seems like a nice clarification: Someone needs to add their node, then it needs to be approved before the Epoch transition.